internal/services/elasticsan/elastic_san_volume_snapshot_data_source.go

```diff
@@ -0,0 +1,147 @@
+package elasticsan
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/elasticsan/2023-01-01/snapshots"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/elasticsan/2023-01-01/volumes"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/elasticsan/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type ElasticSANVolumeSnapshotDataSource struct{}
+
+var _ sdk.DataSource = ElasticSANVolumeSnapshotDataSource{}
+
+type ElasticSANVolumeSnapshotDataSourceModel struct {
+	Name                string                                 `tfschema:"name"`
+	SourceVolumeSizeGiB int64                                  `tfschema:"source_volume_size_gib"`
+	CreateSource        []ElasticSANVolumeSnapshotCreateSource `tfschema:"creation_source"`
+	VolumeGroupId       string                                 `tfschema:"volume_group_id"`
+	VolumeName          string                                 `tfschema:"volume_name"`
+}
+
+type ElasticSANVolumeSnapshotCreateSource struct {
+	SourceId string `tfschema:"source_id"`
+}
+
+func (r ElasticSANVolumeSnapshotDataSource) ResourceType() string {
+	return "azurerm_elastic_san_volume_snapshot"
+}
+
+func (r ElasticSANVolumeSnapshotDataSource) ModelObject() interface{} {
+	return &ElasticSANVolumeSnapshotDataSourceModel{}
+}
+
+func (r ElasticSANVolumeSnapshotDataSource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validate.ElasticSanSnapshotName,
+		},
+
+		"volume_group_id": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+	}
+}
+
+func (r ElasticSANVolumeSnapshotDataSource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"creation_source": {
+			Type:     pluginsdk.TypeList,
+			Computed: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"source_id": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+				},
+			},
+		},
```

Comment:
Can we flatten this to
```suggestion
		"source_id": {
			Type:     pluginsdk.TypeString,
			Computed: true,
		},
```

Reply:
make sense, changed